### ISSUE TITLE
Cloudflare 404 for not excluded files in _routes.json with trailingSlash option

### ISSUE THREAD

**Issue by korywka** [sveltejs/kit#8708]
### Describe the bug

If we have 100+ routes (actually ~4000) for prerender all routes that are NOT in excluded list:
```
"exclude":["/_app/*","/.DS_Store","/favicon.png","/global.css","/vite-manifest.json","/","/__data.json","/bin/vimdiff","/bin/vimdiff/__data.json"
```
like `/bin/cat` gives 404. But we have `bin/cat/index.html` file locally. So prerender works 100% correct. It is not missing file.

Removing `export const trailingSlash = 'always';` seems like fixed that. It was quite easy after finding issue #8422 and `need to check that prerendered pages (with/without trailing slashes) will still work with this setup.` statement :)

### Reproduction

This 4000+ routes are rendered from private SQLite, but I can share this project with some of your developers by email.

### System Info

```Shell
System:
    OS: macOS 13.1
    CPU: (10) arm64 Apple M1 Pro
    Memory: 87.28 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 19.4.0 - /opt/homebrew/bin/node
    npm: 9.2.0 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Safari: 16.2
  npmPackages:
    @sveltejs/adapter-auto: 1.0.2 => 1.0.2 
    @sveltejs/kit: 1.2.6 => 1.2.6 
    svelte: 3.55.1 => 3.55.1 
    vite: 4.0.4 => 4.0.4
```

### Severity

serious, but I can work around it

**Comment by s3812497** [comment]
The behaviour you're looking for can be achieved by changing `prerender` to `'auto'`:
```js
// +page.js
export const prerender = 'auto'; // server will redirect '/bin/cat' to '/bin/cat/'
export const trailingSlash = 'always';
```
or you can include this in the file `/static/_redirects` to use [Cloudflare Pages Redirects](https://developers.cloudflare.com/pages/platform/redirects/):
```
/bin/* /bin/:path/
```

When using `prerender = true`, SvelteKit does not automatically redirect trailing slashes and vice-versa.

With `prerender = true` and:
* `trailingSlash = 'always'` , navigating to `/bin/cat` will result in a 404.
* `trailingSlash = 'never'` (the default), navigating to `/bin/cat/` will result in 404.
* `trailingSlash = 'ignore'`, navigating to `/bin/cat/` will result in 404.

Perhaps this can be documented more thoroughly for people looking to retain the trailing slash redirection.

**Comment by korywka** [comment]
The most weird thing is that ~100 routes work correctly (that are excluded in cf routes config), and ~3900 are 404.

Imho, behavior should be consistent anyway.

**Comment by s3812497** [comment]
> The most weird thing is that ~100 routes work correctly (that are excluded in cf routes config), and ~3900 are 404.

I'd assume that Cloudflare is doing the redirection for the excluded routes.

The rest of the ~3900 are being routed to the SvelteKit Cloudflare function, which doesn't have the routes in its manifest because of `prerender = true`. Changing this to `prerender = 'auto'` ensures they're included.

I set up this [reproduction](https://stackblitz.com/edit/sveltejs-kit-template-default-m14avd?file=src%2Froutes%2Ftest%2F[id]%2F%2Bpage.js) to test the behaviour of the `trailingSlash` option.
Running `npm run build && npm run preview` then navigating to `/test/1` will result in a 404 despite `/test/1/` being in the excluded list in `.svelte-kit/cloudflare/_routes.json`

**Comment by korywka** [comment]
I will fake data and share my project. I need some time. Thank you for your interest.

**Comment by korywka** [comment]
@s3812497 please, take a look at this video:

https://user-images.githubusercontent.com/988471/214717471-f9feca20-9660-4385-bf36-95ddbdc93a85.mov

I am preparing the project.

**Comment by korywka** [comment]
I will not make video without `export const trailingSlash = 'always';` cause it works as expected.

**Comment by s3812497** [comment]
> @s3812497 please, take a look at this video:
> 
>  with-slash.mov 

Interestingly, I can access the prerendered page data https://5f68ec0e.binaries-web.pages.dev/bin/cat/__data.json but not the prerendered page itself https://5f68ec0e.binaries-web.pages.dev/bin/cat/index.html

Now I wonder if removing all the entries in `exclude` for `_routes.json` will cause the other prerendered routes to exhibit the same buggy behaviour.

**Comment by korywka** [comment]
You are right! __data.json is accessible, but not the .html! It is some kind of magic 🙂
